File: src/Java/153. Find Minimum in Rotated Sorted Array.java
class Solution {
    public int findMin(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        int l = 0, r = nums.length - 1;
        while(l < r){
            int mid = l + ((r - l) >> 1);
            if(nums[mid] <= nums[nums.length - 1]){
                r = mid;
            }else{
                l = mid + 1;
            }
        }
        return nums[r];
    }
}

File: src/Go/153. Find Minimum in Rotated Sorted Array.go
func findMin(nums []int) int {
    if len(nums) == 0 {
        return 0;
    }
    l := 0;
    r := len(nums) - 1;
    for l < r {
        mid := (l + r) >> 1;
        if nums[mid] <= nums[len(nums) - 1] {
            r = mid;
        }else{
            l = mid + 1;
        }
    }
    return nums[r];
}